cdc/kv/client.go

```diff
@@ -1192,147 +1283,132 @@ func (s *eventFeedSession) singleEventFeed(
-							zap.String("Event Type", "COMMITTED"),
-							zap.Uint64("CommitTs", entry.CommitTs),
-							zap.Uint64("resolvedTs", checkpointTs),
-							zap.Uint64("regionID", regionID))
-					}
-					select {
-					case s.eventCh <- revent:
-						metricSendEventCommittedCounter.Inc()
-					case <-ctx.Done():
-						return checkpointTs, errors.Trace(ctx.Err())
-					}
-				case cdcpb.Event_PREWRITE:
-					metricPullEventPrewriteCounter.Inc()
-					matcher.putPrewriteRow(entry)
-				case cdcpb.Event_COMMIT:
-					metricPullEventCommitCounter.Inc()
-					if entry.CommitTs <= checkpointTs {
-						log.Fatal("The CommitTs must be greater than the resolvedTs",
-							zap.String("Event Type", "COMMIT"),
-							zap.Uint64("CommitTs", entry.CommitTs),
-							zap.Uint64("resolvedTs", checkpointTs),
-							zap.Uint64("regionID", regionID))
-					}
-					// emit a value
-					value, ok := matcher.matchRow(entry)
-					if !ok {
-						if !initialized {
-							matcher.cacheCommitRow(entry)
-							continue
-						}
-						return checkpointTs,
-							errors.Errorf("prewrite not match, key: %b, start-ts: %d",
-								entry.GetKey(), entry.GetStartTs())
-					}
-
-					revent, err := assembleCommitEvent(entry, value)
-					if err != nil {
-						return checkpointTs, errors.Trace(err)
-					}
-
-					select {
-					case s.eventCh <- revent:
-						metricSendEventCommitCounter.Inc()
-					case <-ctx.Done():
-						return checkpointTs, errors.Trace(ctx.Err())
-					}
-				case cdcpb.Event_ROLLBACK:
-					metricPullEventRollbackCounter.Inc()
-					matcher.rollbackRow(entry)
+				}
+			case *cdcpb.Event_Admin_:
+				log.Info("receive admin event", zap.Stringer("event", event.changeEvent))
+			case *cdcpb.Event_Error:
+				return checkpointTs, errors.Trace(&eventError{err: x.Error})
+			case *cdcpb.Event_ResolvedTs:
+				if err := handleResolvedTs(x.ResolvedTs); err != nil {
+					return checkpointTs, err
 				}
 			}
-		case *cdcpb.Event_Admin_:
-			log.Info("receive admin event", zap.Stringer("event", event))
-		case *cdcpb.Event_Error:
-			return checkpointTs, errors.Trace(&eventError{err: x.Error})
-		case *cdcpb.Event_ResolvedTs:
-			if !initialized {
-				continue
-			}
-			if x.ResolvedTs < checkpointTs {
-				log.Warn("The resolvedTs is fallen back in kvclient",
-					zap.String("Event Type", "RESOLVED"),
-					zap.Uint64("resolvedTs", x.ResolvedTs),
-					zap.Uint64("lastResolvedTs", checkpointTs),
-					zap.Uint64("regionID", regionID))
-				continue
-			}
-			// emit a checkpointTs
-			revent := &model.RegionFeedEvent{
-				Resolved: &model.ResolvedSpan{
-					Span:       span,
-					ResolvedTs: x.ResolvedTs,
-				},
-			}
-			checkpointTs = x.ResolvedTs
+		}
 
-			select {
-			case s.eventCh <- revent:
-				metricSendEventResolvedCounter.Inc()
-			case <-ctx.Done():
-				return checkpointTs, errors.Trace(ctx.Err())
+		if event.resolvedTs != nil {
+			if err := handleResolvedTs(event.resolvedTs.Ts); err != nil {
```

Comment:
The `ResolvedTs` event contains a list of region ids, but it is not used now.
Will the `ResolvedTs` event be sent to each region in this region list, then the count of resolved ts event seems not reduced.

Reply:
Sure, it's for reducing the message from tikv to ticdc.
If we want to reduce the event inside ticdc, the tikv client may need to be carefully refactored.